plutus-core/exe/Main.hs

```diff
@@ -196,6 +204,37 @@ outputformat = option (maybeReader formatReader)
   <> showDefault
   <> help ("Output format: " ++ formatHelp))
 
+
+-- Reader for budget.  The --restricting option requires two integer arguments
+-- and the easiest way to do this is to supply a colon-separated pair of
+-- integers.
+exbudgetReader :: ReadM ExBudget
+exbudgetReader = do
+  s <- str
+  case splitOn ":" s of
+    [a,b] -> case (readMaybe a, readMaybe b) of
+               (Just (cpu::Integer), Just (mem::Integer)) -> pure $ ExBudget (ExCPU cpu) (ExMemory mem)
```

Comment:
Are type annotations needed?

Reply:
Apparently not.  Thanks for spotting that.